### ISSUE TITLE
Chinook movement bug directly after construction plus weird rally point behaviour for all faction's supply centers

### ISSUE THREAD

**Issue by kABUSE1** [TheSuperHackers/GeneralsGamePatch#1985]
If directly after construction (of either chinook or supply center) you order the Chinook to go somewhere else BEFORE it has reached its default flight height, it will go back to collecting supplies once it arrives at the position you have commanded it to fly to.

This also happens if you set a rally point for the supply center. This has to be touched with caution, tho. As you can't undo rally points, it would mean that a supply dock will never auto-collect again after you set it to have a rally point.

The ideal solution would be to give the "Rally Point" command button the functionality of setting a building back to "no rally point" mode. Maybe by right-clicking or whatever. Then we can also finally stop the discussion of whether the rally point command button has a purpose or not, lmao.

An alternative solution would be to add the ability to "select" the supply pile that newly constructed collectors are supposed to collect from via a rally point. This would add another senseful micro action to the game. But I imagine this to cause new issues, as small supply piles completely vanish after being fully collected, so some special cases will require lots of testing...

https://github.com/TheSuperHackers/GeneralsGamePatch/assets/47530719/6245949c-cd5b-47a3-8ca4-2f644522846c

**Comment by MTKing4** [comment]
Theoritcally, Rally points are useful to let a chinook collect from the nearest supply to the rally point, I've not tested that so i don't know if it works that way

But it's useless in most cases because it only works for newly built chinooks, and usually the first 1-2 chinooks will always be built near a supply anyway so Rally point is not needed

But a good feature would be if the rally points can be assigned to the current functioning chinooks at that supply as well as the newly built ones, so that you can order a collection to a new supply without having select all the chinooks, which can be cumbersome at times

And what would also be nice is to be able to actually put the rally point on the supply itself, basically a collection order, and then it'll do its job from then on, but all that for thyme of course

There's also something to keep in mind, i think there's some sort of radius on how further a chinook (other collectors) will look for new supplies once the current one is finished, because sometimes i see them go to new supplies on their own and sometimes they don't, would be nice to know how much is the distance factor for this

**Comment by kABUSE1** [comment]
> There's also something to keep in mind, i think there's some sort of radius on how further a chinook (other collectors) will look for new supplies once the current one is finished, because sometimes i see them go to new supplies on their own and sometimes they don't, would be nice to know how much is the distance factor for this

It's around 4 command center widths that the nearest populated supply pile needs to be within. Same for all factions.

![grafik](https://github.com/TheSuperHackers/GeneralsGamePatch/assets/47530719/67072a53-d593-447b-99cb-37c0855496fc)

**Comment by kABUSE1** [comment]
> Theoritcally, Rally points are useful to let a chinook collect from the nearest supply to the rally point, I've not tested that so i don't know if it works that way

Good point. I totally missed on that.

You are correct, it works as you suspected. They will move to the nearest supply dock you specified.
But IDIOTICALLY, if done through this method, they will pick up and deliver goods only once, if the supply pile they approached ends up being out of range (see above imgage)

**Comment by kABUSE1** [comment]
WTF EA...
https://github.com/TheSuperHackers/GeneralsGamePatch/assets/47530719/f0b7ee2f-9593-4ce2-a5ab-d33a61ef10d6

**Comment by kABUSE1** [comment]
> Theoritcally, Rally points are useful to let a chinook collect from the nearest supply to the rally point, I've not tested that so i don't know if it works that way

Good point. I totally missed on that.

You are correct, it works as you suspected (kinda). Yes, the collector will move to the nearest supply dock you specified via rally point. But IDIOTICALLY, if done through this method, they will pick up and deliver goods from that source only once, unless the dock they reached through the rally point method is (1.) in range and (2.) the closest one in range.

https://github.com/TheSuperHackers/GeneralsGamePatch/assets/47530719/69dc3a8d-e77d-4579-81e5-5f15895b0dd5

Same behaviour (and trigger range) for China and GLA (see above image)

If another source is closer and in range, they will switch to that one after first collection.
If none is in range, they will stop moving after first collection.

So right now there really is pretty much no use-case for the Rally Points of supply centers, only bugs and bullshit.

**Comment by MTKing4** [comment]
Nice findings

So yea we conclude that rally points are only controlling on unit creations, and the only way to make them useful is to control chinooks on drop off as well, similar to how repaired units at warfactories still follow rally points after repair, then we can use rally points as intended, like the video above.

**Comment by xezon** [comment]
Duplicate of TheSuperHackers/GeneralsGameCode#45
